### ISSUE TITLE
failure on udx/test_ucp_tag_mem_type.xfer_mismatch_length

### ISSUE THREAD

**Issue by alinask** [openucx/ucx#4798]
Failure on:
```
11:45:48 [ RUN      ] udx/test_ucp_tag_mem_type.xfer_mismatch_length/12 <ud_x,cuda_copy,rocm_copy>
11:45:48 [     INFO ] TEST: cuda <-> host length :8596213
11:45:52 ==5111== Thread 5:
11:45:52 ==5111== Invalid read of size 8
11:45:52 ==5111==    at 0x4C2E080: memcpy@@GLIBC_2.14 (vg_replace_strmem.c:1018)
11:45:52 ==5111==    by 0x5B139AD: uct_ud_ep_resend (ud_ep.c:969)
11:45:52 ==5111==    by 0x5B139AD: uct_ud_ep_do_pending_ctl (ud_ep.c:1017)
11:45:52 ==5111==    by 0x5B15373: uct_ud_ep_do_pending (ud_ep.c:1140)
11:45:52 ==5111==    by 0x5082DD2: ucs_arbiter_dispatch_nonempty (arbiter.c:286)
11:45:52 ==5111==    by 0x5B1BB13: ucs_arbiter_dispatch (arbiter.h:356)
11:45:52 ==5111==    by 0x5B1BB13: uct_ud_iface_progress_pending (ud_iface.h:397)
11:45:52 ==5111==    by 0x5B1BB13: uct_ud_mlx5_iface_async_progress (ud_mlx5.c:504)
11:45:52 ==5111==    by 0x5B0EBDA: uct_ud_iface_async_progress (ud_iface.c:849)
11:45:52 ==5111==    by 0x5B0EBDA: uct_ud_iface_timer (ud_iface.c:864)
11:45:52 ==5111==    by 0x5079C56: ucs_async_handler_invoke (async.c:224)
11:45:52 ==5111==    by 0x5079C56: ucs_async_handler_dispatch (async.c:242)
11:45:52 ==5111==    by 0x5079C56: ucs_async_dispatch_handlers (async.c:271)
11:45:52 ==5111==    by 0x5079FE0: ucs_async_dispatch_timerq (async.c:298)
11:45:52 ==5111==    by 0x507DAD4: ucs_async_thread_func (thread.c:138)
11:45:52 ==5111==    by 0x79CAE24: start_thread (in /usr/lib64/libpthread-2.17.so)
11:45:52 ==5111==    by 0x8D3F34C: clone (in /usr/lib64/libc-2.17.so)
11:45:52 ==5111==  Address 0x3a3e81b0 is in a --- anonymous segment
11:45:52 ==5111== 
11:45:52 [hpc-test-node-gpu02:5111 :0:7814] Caught signal 11 (Segmentation fault: invalid permissions for mapped object at address 0x3a3e81b0)
11:45:59 ==== backtrace (tid:   7814) ====
11:45:59  0 0x0000000000054f38 ucs_debug_print_backtrace()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node-gpu/worker/0/contrib/../src/ucs/debug/debug.c:625
11:45:59  1 0x0000000000008080 _vgr20180ZZ_libcZdsoZa_memcpyZAZAGLIBCZu2Zd14()  /builddir/build/BUILD/valgrind-3.12.0/memcheck/../shared/vg_replace_strmem.c:1018
11:45:59  2 0x000000000006b9ae uct_ud_ep_resend()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node-gpu/worker/0/contrib/../src/uct/ib/ud/base/ud_ep.c:969
11:45:59  3 0x000000000006d374 uct_ud_ep_do_pending()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node-gpu/worker/0/contrib/../src/uct/ib/ud/base/ud_ep.c:1140
11:45:59  4 0x000000000004add3 ucs_arbiter_dispatch_nonempty()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node-gpu/worker/0/contrib/../src/ucs/datastruct/arbiter.c:286
11:45:59  5 0x0000000000073b14 ucs_arbiter_dispatch()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node-gpu/worker/0/contrib/../src/ucs/datastruct/arbiter.h:356
11:45:59  6 0x0000000000066bdb uct_ud_iface_async_progress()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node-gpu/worker/0/contrib/../src/uct/ib/ud/base/ud_iface.c:849
11:45:59  7 0x0000000000041c57 ucs_async_handler_invoke()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node-gpu/worker/0/contrib/../src/ucs/async/async.c:224
11:45:59  8 0x0000000000041fe1 ucs_async_dispatch_timerq()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node-gpu/worker/0/contrib/../src/ucs/async/async.c:298
11:45:59  9 0x0000000000045ad5 ucs_async_thread_func()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node-gpu/worker/0/contrib/../src/ucs/async/thread.c:138
11:45:59 10 0x0000000000007e25 start_thread()  pthread_create.c:0
11:45:59 11 0x00000000000f834d __clone()  ???:0
11:45:59 =================================
```
on hpc-test-node-gpu02

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/14637/label=hpc-test-node-gpu,worker=0/console

failed when running with valgrind

**Comment by yosefe** [comment]
missing gpu mem support in UD for resend flow

**Comment by yosefe** [comment]
fixed by #4958
